server/chat/backend/agent/tools/cloud_tools.py

```
@@ -686,45 +686,52 @@ def wrapper(*args, **kwargs):
+
         # ALWAYS get session_id from context, never from AI
         context_state = get_state_context()
         context_session_id = context_state.session_id if context_state and hasattr(context_state, 'session_id') else None
         if context_session_id:
-            kwargs['session_id'] = context_session_id
-            logging.info(f"with_forced_context: Forced session_id from context: {context_session_id} for {func.__name__}")
+            if 'session_id' in func_params:
+                kwargs['session_id'] = context_session_id
+                logging.info(f"with_forced_context: Forced session_id from context: {context_session_id} for {func.__name__}")
         else:
             logging.error(f"with_forced_context: No session_id in context for {func.__name__}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)